File: golang/q03/part1/part1.go
package part1

import (
	"aoc2023/q03/common"
	"fmt"
)

func Part1(lines []string) {
	schematic := common.Parse(lines)
	sum := 0

	fmt.Printf("%+v\n", schematic.Numbers)

	for _, part := range schematic.Numbers {
		neighbours := schematic.NeighborSymbols(part)
		if len(neighbours) > 0 {
			sum += part.Value
		}
	}

	fmt.Println("Part 1:", sum)
}

File: js/q03/part1/part1.js
import { getSymbolNeighbours, parseFile } from "../common/common.js";

function part1(path) {
  const parseResult = parseFile(path);
  if (parseResult.isErr()) {
    console.error(parseResult.unwrap());
    return;
  }

  const schema = parseResult.unwrap();
  let sum = 0;

  for (const partNumber of schema.numbers) {
    const neighbours = getSymbolNeighbours(partNumber, schema.symbols)
    if (neighbours.length > 0) {
      sum += parseInt(partNumber.number);
    }
  }

  console.log(`Part 1: ${sum}`);
}

part1('./data/input.txt');
